app/controllers/admin/order_cycles_controller.rb

```diff
@@ -299,5 +295,23 @@ def order_cycle_bulk_params
         collection_attributes: [:id] + PermittedAttributes::OrderCycle.basic_attributes
       ).to_h.with_indifferent_access
     end
+
+    # Check that order cycle datetime values changed if it has existing orders
+    def verify_datetime_change
+      return unless params[:order_cycle][:confirm]
+      return unless @order_cycle.orders.exists?
+      return if same_dates(@order_cycle.orders_open_at&.to_s,
```

Comment:
`@order_cycle.orders_open_at` is `ActiveSupport::TimeWithZone` so can just do `@order_cycle.orders_open_at.to_datetime`  , no need to cast to string to then re parse it.
Same with `@order_cycle.orders_close_at` below.

Could you rewrite this to remove the unnecessary casting/parsing ?